### PR TITLE
[MNT] or [EHN] (?) Exposing EEGITNet parameters

### PR DESCRIPTION
**Pull request by bruAristimunha**
I'm working on normalizing the parameters related to the modelz to improve the search for hyperparameters, and I noticed that this model has no exposed parameters.

Related with #649

### CODE REVIEW

File: braindecode/models/eegitnet.py [hunk 1]
```diff
@@ -357,3 +375,11 @@ def _get_inception_branch(
...
             activation(),
         )
+
+
+if __name__ == "__main__":
+    x = torch.zeros(1, 22, 1001)
+    model = EEGITNet(n_chans=22, n_outputs=2, n_times=1001, sfreq=256)
+
+    out = model(x)
+    print(out.shape)
```
Comment (bruAristimunha) [inline braindecode/models/eegitnet.py:385]:
```suggestion
```

File: braindecode/models/eegitnet.py [hunk 2]
```diff
@@ -271,10 +294,10 @@ def __init__(
...
             "TC_block2",
             _TCBlock(
-                in_ch=14,
-                kernel_length=4,
-                dialation=2,
-                padding=6,
+                in_ch=tcn_in_channel,
+                kernel_length=tcn_kernel_size,
+                dilatation=int(tcn_dilatation * 2),
+                padding=int(tcn_padding * 2),
```
Comment (PierreGtch) [inline braindecode/models/eegitnet.py:300]:
```suggestion
                dilatation=tcn_dilatation * 2,
                padding=tcn_padding * 2,
```
it should not be needed if you multiply :) 
Same below

File: braindecode/models/eegitnet.py [hunk 3]
```diff
@@ -283,10 +306,10 @@ def __init__(
...
             "TC_block3",
             _TCBlock(
-                in_ch=14,
-                kernel_length=4,
-                dialation=4,
-                padding=12,
+                in_ch=tcn_in_channel,
+                kernel_length=tcn_kernel_size,
+                dilatation=int(tcn_dilatation * 4),
+                padding=int(tcn_padding * 4),
```
Comment (PierreGtch) [inline braindecode/models/eegitnet.py:312]:
```suggestion
                dilatation=tcn_dilatation * 4,
                padding=tcn_padding * 4,
```

File: braindecode/models/eegitnet.py [hunk 4]
```diff
@@ -295,10 +318,10 @@ def __init__(
...
             "TC_block4",
             _TCBlock(
-                in_ch=14,
-                kernel_length=4,
-                dialation=8,
-                padding=24,
+                in_ch=tcn_in_channel,
+                kernel_length=tcn_kernel_size,
+                dilatation=int(tcn_dilatation * 8),
+                padding=int(tcn_padding * 8),
```
Comment (PierreGtch) [inline braindecode/models/eegitnet.py:324]:
```suggestion
                dilatation=tcn_dilatation * 8,
                padding=tcn_padding * 8,
```

File: braindecode/models/eegitnet.py [hunk 5]
```diff
@@ -308,27 +331,22 @@ def __init__(
         self.add_module(
             "dim_reduction",
             nn.Sequential(
-                nn.Conv2d(14, 28, kernel_size=(1, 1)),
-                nn.BatchNorm2d(28),
+                nn.Conv2d(tcn_in_channel, int(tcn_in_channel * 2), kernel_size=(1, 1)),
+                nn.BatchNorm2d(int(tcn_in_channel * 2)),
```
Comment (PierreGtch) [inline braindecode/models/eegitnet.py:335]:
```suggestion
                nn.Conv2d(tcn_in_channel, tcn_in_channel * 2, kernel_size=(1, 1)),
                nn.BatchNorm2d(tcn_in_channel * 2),
```

File: braindecode/models/eegitnet.py [hunk 6]
```diff
@@ -234,35 +255,37 @@ def __init__(
...
+            out_channels=int(n_filters_time * 2),
+            kernel_length=int(kernel_length * 2),
             activation=activation,
         )
         block13 = self._get_inception_branch(
             in_channels=self.n_chans,
-            out_channels=8,
-            kernel_length=64,
+            out_channels=int(n_filters_time * 4),
+            kernel_length=int(n_filters_time * 4),
```
Comment (PierreGtch) [inline braindecode/models/eegitnet.py:271]:
```suggestion
            out_channels=n_filters_time * 4,
            kernel_length=n_filters_time * 4,
```

File: braindecode/models/eegitnet.py [hunk 7]
```diff
@@ -234,35 +255,37 @@ def __init__(
...
+            out_channels=n_filters_time,
+            kernel_length=kernel_length,
             activation=activation,
         )
         block12 = self._get_inception_branch(
             in_channels=self.n_chans,
-            out_channels=4,
-            kernel_length=32,
+            out_channels=int(n_filters_time * 2),
+            kernel_length=int(kernel_length * 2),
```
Comment (PierreGtch) [inline braindecode/models/eegitnet.py:265]:
```suggestion
            out_channels=n_filters_time * 2,
            kernel_length=kernel_length * 2,
```

File: braindecode/models/eegitnet.py [hunk 8]
```diff
@@ -308,27 +331,22 @@ def __init__(
...
         # Moved flatten to another layer
         self.add_module("flatten", nn.Flatten())
 
-        # Incorporating classification module and subsequent ones in one final layer
-        module = nn.Sequential()
-
-        module.add_module(
-            "clf", nn.Linear(int(int(self.n_times / 4) / 4) * 28, self.n_outputs)
+        num_features = int(int(self.n_times / tcn_kernel_size) / 4) * int(
+            tcn_in_channel * 2
```
Comment (PierreGtch) [inline braindecode/models/eegitnet.py:346]: Are you sure about the math?
 
The safe way would be to use:
```suggestion
        num_features = self.get_output_shape()
```
But it would make the model initialisation a bit longer...

**Review by PierreGtch — CHANGES_REQUESTED** [review]
One comment and one question, otherwise LGTM :)

File: braindecode/models/eegitnet.py [hunk 9]
```diff
@@ -308,27 +331,21 @@ def __init__(
...
         # Moved flatten to another layer
         self.add_module("flatten", nn.Flatten())
 
-        # Incorporating classification module and subsequent ones in one final layer
-        module = nn.Sequential()
-
-        module.add_module(
-            "clf", nn.Linear(int(int(self.n_times / 4) / 4) * 28, self.n_outputs)
+        num_features = self.get_output_shape()
         )
```
Comment (bruAristimunha) [inline braindecode/models/eegitnet.py:346]:
```suggestion
        num_features = self.get_output_shape()
```

File: braindecode/models/eegitnet.py [hunk 10]
```diff
@@ -308,27 +331,20 @@ def __init__(
...
 
-        # Incorporating classification module and subsequent ones in one final layer
-        module = nn.Sequential()
-
-        module.add_module(
-            "clf", nn.Linear(int(int(self.n_times / 4) / 4) * 28, self.n_outputs)
-        )
-
-        module.add_module("out_fun", nn.Identity())
+        num_features = self.get_output_shape()
```
Comment (bruAristimunha) [inline braindecode/models/eegitnet.py:345]:
```suggestion
        num_features = self.get_output_shape()[-1]
```
